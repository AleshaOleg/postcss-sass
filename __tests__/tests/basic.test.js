const getPostCssTreeFromSass = require('../../helpers/getPostCssTreeFromSassTree');

it('basic.sass', function () {
    expect(getPostCssTreeFromSass('basic')).toMatchSnapshot();
});
