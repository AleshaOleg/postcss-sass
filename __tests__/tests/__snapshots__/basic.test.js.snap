// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic.sass 1`] = `
Object {
  "nodes": Array [
    Object {
      "nodes": Array [
        Object {
          "prop": "font",
          "raws": Object {
            "before": "
    :",
            "between": " ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 15,
              "line": 4,
            },
            "input": Input {
              "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 4,
            },
          },
          "type": "decl",
          "value": "Arial",
        },
        Object {
          "prop": "background",
          "raws": Object {
            "before": "
    :",
            "between": " ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 20,
              "line": 5,
            },
            "input": Input {
              "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 5,
            },
          },
          "type": "decl",
          "value": "blue",
        },
      ],
      "raws": Object {
        "before": "

",
        "between": "",
      },
      "selector": "body",
      "source": Object {
        "end": Object {
          "column": 21,
          "line": 5,
        },
        "input": Input {
          "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 1,
          "line": 3,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "width",
          "raws": Object {
            "before": "
    :",
            "between": " ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 16,
              "line": 8,
            },
            "input": Input {
              "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 8,
            },
          },
          "type": "decl",
          "value": "700px",
        },
        Object {
          "prop": "height",
          "raws": Object {
            "before": "
    :",
            "between": " ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 15,
              "line": 9,
            },
            "input": Input {
              "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 9,
            },
          },
          "type": "decl",
          "value": "100",
        },
        Object {
          "nodes": Array [
            Object {
              "prop": "height",
              "raws": Object {
                "before": "
        :",
                "between": " ",
                "semicolon": false,
              },
              "source": Object {
                "end": Object {
                  "column": 21,
                  "line": 12,
                },
                "input": Input {
                  "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
                  "id": "<input css 1>",
                },
                "start": Object {
                  "column": 9,
                  "line": 12,
                },
              },
              "type": "decl",
              "value": "300px",
            },
            Object {
              "nodes": Array [
                Object {
                  "prop": "font-size",
                  "raws": Object {
                    "before": "
            :",
                    "between": " ",
                    "semicolon": false,
                  },
                  "source": Object {
                    "end": Object {
                      "column": 27,
                      "line": 14,
                    },
                    "input": Input {
                      "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
                      "id": "<input css 1>",
                    },
                    "start": Object {
                      "column": 13,
                      "line": 14,
                    },
                  },
                  "type": "decl",
                  "value": "50px",
                },
                Object {
                  "prop": "color",
                  "raws": Object {
                    "before": "
            :",
                    "between": "     ",
                    "semicolon": false,
                  },
                  "source": Object {
                    "end": Object {
                      "column": 27,
                      "line": 15,
                    },
                    "input": Input {
                      "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
                      "id": "<input css 1>",
                    },
                    "start": Object {
                      "column": 13,
                      "line": 15,
                    },
                  },
                  "type": "decl",
                  "value": "blue",
                },
              ],
              "raws": Object {
                "before": "
        ",
                "between": "",
              },
              "selector": "h1",
              "source": Object {
                "end": Object {
                  "column": 28,
                  "line": 15,
                },
                "input": Input {
                  "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
                  "id": "<input css 1>",
                },
                "start": Object {
                  "column": 9,
                  "line": 13,
                },
              },
              "type": "rule",
            },
          ],
          "raws": Object {
            "before": "

    ",
            "between": "",
          },
          "selector": "#header",
          "source": Object {
            "end": Object {
              "column": 1,
              "line": 16,
            },
            "input": Input {
              "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 11,
            },
          },
          "type": "rule",
        },
      ],
      "raws": Object {
        "before": "

",
        "between": "",
      },
      "selector": "#page",
      "source": Object {
        "end": Object {
          "column": 1,
          "line": 16,
        },
        "input": Input {
          "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 1,
          "line": 7,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "nodes": Array [
            Object {
              "nodes": Array [
                Object {
                  "prop": "width",
                  "raws": Object {
                    "before": "
            :",
                    "between": " ",
                    "semicolon": false,
                  },
                  "source": Object {
                    "end": Object {
                      "column": 24,
                      "line": 20,
                    },
                    "input": Input {
                      "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
                      "id": "<input css 1>",
                    },
                    "start": Object {
                      "column": 13,
                      "line": 20,
                    },
                  },
                  "type": "decl",
                  "value": "100px",
                },
              ],
              "raws": Object {
                "before": "
        ",
                "between": "",
              },
              "selector": "#column.left",
              "source": Object {
                "end": Object {
                  "column": 25,
                  "line": 20,
                },
                "input": Input {
                  "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
                  "id": "<input css 1>",
                },
                "start": Object {
                  "column": 9,
                  "line": 19,
                },
              },
              "type": "rule",
            },
            Object {
              "nodes": Array [
                Object {
                  "prop": "width",
                  "raws": Object {
                    "before": "
            :",
                    "between": " ",
                    "semicolon": false,
                  },
                  "source": Object {
                    "end": Object {
                      "column": 24,
                      "line": 22,
                    },
                    "input": Input {
                      "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
                      "id": "<input css 1>",
                    },
                    "start": Object {
                      "column": 13,
                      "line": 22,
                    },
                  },
                  "type": "decl",
                  "value": "600px",
                },
              ],
              "raws": Object {
                "before": "
        ",
                "between": "",
              },
              "selector": "#column.right",
              "source": Object {
                "end": Object {
                  "column": 25,
                  "line": 22,
                },
                "input": Input {
                  "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
                  "id": "<input css 1>",
                },
                "start": Object {
                  "column": 9,
                  "line": 21,
                },
              },
              "type": "rule",
            },
          ],
          "raws": Object {
            "before": "
    ",
            "between": "",
          },
          "selector": "#container.top",
          "source": Object {
            "end": Object {
              "column": 4,
              "line": 23,
            },
            "input": Input {
              "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 18,
            },
          },
          "type": "rule",
        },
        Object {
          "nodes": Array [
            Object {
              "prop": "background",
              "raws": Object {
                "before": "
        :",
                "between": " ",
                "semicolon": false,
              },
              "source": Object {
                "end": Object {
                  "column": 25,
                  "line": 24,
                },
                "input": Input {
                  "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
                  "id": "<input css 1>",
                },
                "start": Object {
                  "column": 9,
                  "line": 24,
                },
              },
              "type": "decl",
              "value": "brown",
            },
          ],
          "raws": Object {
            "before": "
    ",
            "between": "",
          },
          "selector": "#container.bottom",
          "source": Object {
            "end": Object {
              "column": 26,
              "line": 24,
            },
            "input": Input {
              "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 23,
            },
          },
          "type": "rule",
        },
      ],
      "raws": Object {
        "before": "

",
        "between": "",
      },
      "selector": "#content.user.show",
      "source": Object {
        "end": Object {
          "column": 26,
          "line": 24,
        },
        "input": Input {
          "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 1,
          "line": 17,
        },
      },
      "type": "rule",
    },
  ],
  "raws": Object {
    "before": "",
    "semicolon": undefined,
  },
  "source": Object {
    "end": Object {
      "column": 26,
      "line": 24,
    },
    "input": Input {
      "css": "

body
    :font Arial
    :background blue

#page
    :width 700px
    :height 100

    #header
        :height 300px
        h1
            :font-size 50px
            :color     blue

#content.user.show
    #container.top
        #column.left
            :width 100px
        #column.right
            :width 600px
    #container.bottom
        :background brown
",
      "id": "<input css 1>",
    },
    "start": Object {
      "column": 1,
      "line": 1,
    },
  },
  "type": "root",
}
`;
